src/env/__init_tls.c

```diff
@@ -7,111 +7,155 @@
 #include "libc.h"
 #include "atomic.h"
 #include "syscall.h"
-#include "enclave/sgxlkl_config.h"
-#include "enclave/lthread.h"
-#include "enclave/enclave_util.h"
-
-static int spawned_ethreads = 1;
-
-struct pthread
-{
-};
+#include <enclave/enclave_oe.h>
+#include <enclave/enclave_util.h>
 
 int __init_tp(void *p)
 {
-	struct schedctx *td = p;
-
+	pthread_t td = p;
 	td->self = td;
-	// Prevent collisions with lthread TIDs which are assigned to newly spawned
-	// lthreads incrementally, starting from one.
-	td->tid = INT_MAX - a_fetch_add(&spawned_ethreads, 1);
-	td->locale = &libc.global_locale;
-	td->robust_list.head = &td->robust_list.head;
-	libc.can_do_threads = 1;
-	return 0;
-}
-
-int __init_utp(void *p, int set_tp)
-{
-	struct lthread_tcb_base *tcb = (struct lthread_tcb_base *)p;
-	tcb->self = p;
-	tcb->schedctx = __scheduler_self();
-	if (libc.user_tls_enabled && set_tp)
+	if (sgxlkl_enclave->mode == SW_DEBUG_MODE)
```

Comment:
We won’t be exposing the sgxlkl_enclave symbol to userspace soon, so this needs to do the syscall and have that logic handled in the kernel.